src/Lean/Data/Lsp/Capabilities.lean

```diff
@@ -29,8 +29,17 @@ structure TextDocumentClientCapabilities where
   completion? : Option CompletionClientCapabilities
   deriving ToJson, FromJson
 
+structure ShowDocumentClientCapabilities where
+  support : Bool
+  deriving ToJson, FromJson
+
+structure WindowClientCapabilities where
+  showDocument? : Option ShowDocumentClientCapabilities
+  deriving ToJson, FromJson
+
 structure ClientCapabilities where
   textDocument? : Option TextDocumentClientCapabilities
+  window? : Option WindowClientCapabilities
```

Comment:
It is, I can remove it if you prefer. I thought it would be useful if RPC handlers could know whether the `showDocument` request is available (I used it in a prototype).